src/function.rs

```diff
@@ -127,7 +114,30 @@ impl<'instance> Function<'instance> {
         // Maintain sufficient allocated space in the vector rather than just declaring its capacity.
         let result_count =
             unsafe { wasm_func_get_result_count(self.function, instance.get_inner_instance()) };
-        let capacity = std::cmp::max(param_count, result_count) as usize * 2;
+        let mut result_types = vec![0u8; result_count as usize];
```

Comment:
What we require is a sufficiently large space; there's no need to expend resources on precise calculations. Perhaps `*4` would be adequate.